type.go

```diff
@@ -547,6 +1049,18 @@ func (typeList) Kind() Kind {
 	return KindList
 }
 
+func (t typeList) Equals(other Type) bool {
+	a, err := surge.ToBinary(t)
+	if err != nil {
+		return false
+	}
+	b, err := surge.ToBinary(other)
+	if err != nil {
+		return false
+	}
+	return bytes.Equal(a, b)
```

Comment:
A slightly more complex scenario, but nonetheless, reflection would be better here.

```suggestion
    otherList, ok := other.(typeList)
    if !ok {
        return false
    }
    return t.Type.Equals(otherList.Type)
}
```